notify/notify.go

```diff
@@ -797,6 +830,15 @@ func (r RetryStage) exec(ctx context.Context, l log.Logger, alerts ...*types.Ale
 			if err != nil {
 				r.metrics.numNotificationRequestsFailedTotal.WithLabelValues(r.labelValues...).Inc()
 				if !retry {
+					if r.alertLCObserver != nil {
+						m := alertobserver.AlertEventMeta{
+							"ctx":         ctx,
+							"msg":         "Unrecoverable error",
+							"integration": r.integration.Name(),
```

Comment:
do we care about each retry? should we just record final fail or final success  at func (r RetryStage) Exec()

Reply:
I don't think we should care about retry here, currently we only record the final fail.

Reply:
i mean why not put this into line 758 Exec function

Reply:
I updated the code the log the `sent` alerts instead because it is the correct list of alerts that was sent. I think because we don't return the `sent` alerts we have to keep the code where it currently is.